apps/opt.c

```diff
@@ -326,6 +326,30 @@ int opt_int(const char *value, int *result)
     return 1;
 }
 
+static STRSTR_PAIR base[] = {
+    {"0x", "hex"},
+    {"0X", "hex"},
+    {"0", "oct"},
+    {NULL}
+};
+
+#define opt_number_error(v, b) \
+    do { \
+        int i = 0; \
+        while (b[i].prefix != NULL) { \
+            if (strncmp(v, b[i].prefix, strlen(b[i].prefix)) == 0) { \
+                BIO_printf(bio_err, \
+                           "%s: Can't parse \"%s\" as a number in %s\n", \
```

Comment:
I'm not native English speaker, but does "number in X" sound right? Shouldn't it rather be "X number"? And don't hesitate to write non-lingo, especially when it means something. I mean "hex" is actually a lingo for "hexadecimal", but it's also an actual word...